packages/dev/core/src/FlowGraph/flowGraphInteger.ts

```diff
@@ -0,0 +1,90 @@
+import { RegisterClass } from "../Misc/typeStore";
+
+/**
+ * @experimental
+ * Class that represents an integer value.
+ */
+export class FlowGraphInteger {
+    /**
+     * The value of the integer. Its type
+     * is a javascript number.
+     */
+    value: number;
+
+    constructor(value: number) {
+        this.value = this.convertToInt(value);
+    }
+
+    /**
+     * Converts a float to an integer.
+     * @param n the float to convert
+     * @returns
+     */
+    public convertToInt(n: number): number {
+        return n | 0;
+    }
+
+    /**
+     * Adds two integers together.
+     * @param other the other integer to add
+     * @returns
+     */
+    public add(other: FlowGraphInteger): FlowGraphInteger {
+        return new FlowGraphInteger(this.value + other.value);

```

Comment:
A rather philosophical question - shouldn't we use the toInt values of the two Integers? I mean, let's theoretically say we have two values, 5.7 and 4.8 - Their `toInt` values are 5 and 4, but adding them as values will give 10.5, which will be 10 when using `toInt`.
i guess it is only possible as long as value is both public and not mutable. Making value readonly will probably solve this (for all other operations as well)